README.md

```diff
@@ -1,74 +1,187 @@
-puppet-module-cgroups
-===
+# puppet-module-cgroups
 
-Puppet module to manage cgroups.
+Manage cgroups configuration service and files.
 
-[![Build Status](https://travis-ci.org/Ericsson/puppet-module-cgroups.png?branch=master)](https://travis-ci.org/Ericsson/puppet-module-cgroups)
+- /etc/cgconfig.conf
+- /etc/cgconfig.d/*.conf
 
-# Compatability
+# Compatibility
 
 This module has been tested to work on the following systems with Puppet v3
-(with and without the future parser) and Puppet v4 with Ruby versions 1.8.7,
-1.9.3, 2.0.0 and 2.1.0.
+(with and without the future parser) and Puppet v4 (with strict variables)
+using Ruby versions 1.8.7 (Puppet v3 only), 1.9.3, 2.0.0 and 2.1.0.
 
   * EL 6
+  * EL 7
   * SLED 11 SP2
+  * SLES 11 SP2
+
+[![Build Status](https://travis-ci.org/Ericsson/puppet-module-cgroups.png?branch=master)](https://travis-ci.org/Ericsson/puppet-module-cgroups)
+
+## Class `cgroup`
 
-# Parameters
+### Description
 
-config_file_path
-----------------
-Path to cgroups config file.
+The `cgroups` class is used to configure the cgroup service and its main configuration details.
+
+### Parameters
+
+---
+#### cgconfig_content (string)
+Optional, specify arbitary content for the cgconfig.conf file that will be included at the bottom.
+
+- *Default*: undef
+
+---
+#### config_file_path (string)
+Absolute path to cgroups config file.
 
 - *Default*: '/etc/cgconfig.conf'
 
-service_name
-------------
-name of service.
+---
+#### groups (hash)
+A hash containing group resources to be configured (see below for cgroups::group resources)
+
+- *Default*: undef
+
+##### Example:
+```yaml
+cgroups::groups:
+  "user/mgw-all":
+    permissions:
+      task:
+        uid: root
+        gid: mgw-all
+      admin:
+        uid: root
+        gid: mgw-all
+    controllers:
+      cpuset:
+        "cpuset.mems": "0"
+        "cpuset.cpus": "0,1"
+```
+
+##### Hiera example with Suse 11.2 bugfix:
+```yaml
+cgroups::user_path_fix: '/sys/fs/cgroup/user/mgw-all'
+cgroups::groups:
+  "user/mgw-all":
+    permissions:
+      task:
+        uid: root
+        gid: mgw-all
+      admin:
+        uid: root
+        gid: mgw-all
+    controllers:
+      cpuset:
+        "cpuset.mems": "0"
+        "cpuset.cpus": "0,1"
+```
+---
+#### mounts (hash)
+A hash containing mounts to be configured in /etc/cgconfig.conf.
+NOTE: As of RedHat 7 default resource controllers are the domain of systemd therefore this setting should probably not be used unless you are managing a controller not yet supported by systemd such as net_prio.
```

Comment:
> As of RedHat 7..

I'm much more of a RedHat CentOS person than a Debian user, but I'm reasonably sure that this would apply to any systemd enabled system, not just RedHat 7.... but I need to be fact checked on that one ;-)